src/python/pants/backend/python/rules/coverage.py

```diff
@@ -323,43 +325,66 @@ async def generate_coverage_report(
         ),
     )
 
-    report_type = coverage_subsystem.report
-    process = coverage_setup.pex.create_process(
-        pex_path=f"./{coverage_setup.pex.output_filename}",
-        # We pass `--ignore-errors` because Pants dynamically injects missing `__init__.py` files
-        # and this will cause Coverage to fail.
-        pex_args=(report_type.report_name, "--ignore-errors"),
-        input_digest=input_digest,
-        output_directories=("htmlcov",),
-        output_files=("coverage.xml",),
-        description=f"Generate Pytest {report_type.report_name} coverage report.",
-        python_setup=python_setup,
-        subprocess_encoding_environment=subprocess_encoding_environment,
+    processes = []
+    report_types = coverage_subsystem.reports
+    for report_type in report_types:
+        processes.append(
+            coverage_setup.pex.create_process(
+                pex_path=f"./{coverage_setup.pex.output_filename}",
+                # We pass `--ignore-errors` because Pants dynamically injects missing `__init__.py` files
+                # and this will cause Coverage to fail.
+                pex_args=(report_type.report_name, "--ignore-errors"),
+                input_digest=input_digest,
+                output_directories=("htmlcov",),
+                output_files=("coverage.xml",),
```

Comment:
This works because `output_{directories,files}` does not error if those files are not found. But that's an implementation detail that isn't obvious, e.g. the bug we were seeing last week about coverage data not being generated.

Consider replacing with this:
```suggestion
                output_directories=("htmlcov",) if report_type == CoverageReportType.HTML else None,
                output_files=("coverage.xml",) if report_type == CoverageReportType.XML else None,
```